src/map/ai/controllers/mob_controller.cpp

```diff
@@ -790,6 +791,7 @@ void CMobController::DoRoamTick(time_point tick)
                 else if (PMob->m_roamFlags & ROAMFLAG_EVENT)
                 {
                     // allow custom event action
+                    PMob->PAI->EventHandler.triggerListener("ROAM_TICK", PMob);
```

Comment:
I'm not 100% sure on this, but I think we only needed the one ROAM_TICK down where OnMobRoam is - I think OnMobRoamAction is in addition to that one.  But.. I'm not 100% sure.  Could you try adding a listener to OnMobRoam to a mob with ROAMFLAG_EVENT, and see if it prints more than once per 3sec?

Reply:
And if it does, remove this one, and make sure that the same listener still prints once every 3 sec

Reply:
that went a little over my head. paging @ShelbyZ for backup

Reply:
Remove this line (794), so we avoid firing ROAM_TICK twice for mobs who have ROAMFLAG_EVENT (bcnm, etc) and it should be fine.

Reply:
done